### ISSUE TITLE
A question about focal loss

### ISSUE THREAD

**Issue by LaoYang1994** [open-mmlab/mmdetection#1699]
https://github.com/open-mmlab/mmdetection/blob/4d84161f142b7500089b0db001962bbc07aa869d/mmdet/ops/sigmoid_focal_loss/src/sigmoid_focal_loss_cuda.cu#L49-L52

Why are these lines in this form?

**Comment by Kaeseknacker** [comment]
Because everything in a line would be worse to read? :-)

**Comment by ZwwWayne** [comment]
We follow the [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html).

**Comment by LaoYang1994** [comment]
@ZwwWayne hi, I don't argue this point. My question is:
"term2" is about the negative loss. 
```math
1 - p = \frac{e^{-logit}}{1 + e^{-logit}} = \frac{1}{1 + e^{logit}}
```
the code of line49 ~ 52 adpot different forms according to the sign of logit. I want to know the reason.

**Comment by hellock** [comment]
@LaoYang1994 It is a common implementation trick for softmax or related formulas to avoid overflow.

**Comment by LaoYang1994** [comment]
@hellock 
This is the point that confuses me. I don't know why line 43 isn't in this form?
https://github.com/open-mmlab/mmdetection/blob/4d84161f142b7500089b0db001962bbc07aa869d/mmdet/ops/sigmoid_focal_loss/src/sigmoid_focal_loss_cuda.cu#L43

**Comment by alexwq100** [comment]
@LaoYang1994 , have you know the reason why the code adopt this form?

**Comment by alexwq100** [comment]
@hellock ,看了fcos的实现
计算cls loss的时候：对于一个点,coco为例，预测是一个(1,80) shape的tensor，label是一个0到80之间的整数。我看用的就是上面的SigmoidFocalLossForward，到底是怎么计算的？

万分感谢

**Comment by CoinCheung** [comment]
@hellock Hi, would you tell me why you use that trick to compute `log(1-p)` but you did not use same trick to compute `log(p)` ?
